edgelet/edgelet-core/src/settings.rs

```diff
@@ -459,6 +464,59 @@ impl Listen {
     pub fn management_uri(&self) -> &Url {
         &self.management_uri
     }
+
+    pub fn min_tls_version(&self) -> Option<&Protocol> {
+        self.min_tls_version.as_ref()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Protocol {
+    Tls10,
+    Tls11,
+    Tls12,
+}
+
+impl Display for Protocol {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Protocol::Tls10 => write!(f, "TLS 1.0"),
+            Protocol::Tls11 => write!(f, "TLS 1.1"),
+            Protocol::Tls12 => write!(f, "TLS 1.2"),
+        }
+    }
+}
+
+impl FromStr for Protocol {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_ref() {
+            "tls" | "tls1" | "tls10" | "tlsv10" => Ok(Protocol::Tls10),
+            "tls11" | "tlsv11" => Ok(Protocol::Tls11),
+            "tls12" | "tlsv12" => Ok(Protocol::Tls12),
+            _ => Err(format!("Unsupported TLS protocol version: {}", s)),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for Protocol {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
```

Comment:
`let s = deserializer.deserialize_str()?;`

Reply:
deserialize_str expects a visitor. Why it is better than this?
```rust
impl<'de> Deserialize<'de> for String {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>,
    {
        deserializer.deserialize_string(StringVisitor)
    }

    fn deserialize_in_place<D>(deserializer: D, place: &mut Self) -> Result<(), D::Error>
    where
        D: Deserializer<'de>,
    {
        deserializer.deserialize_string(StringInPlaceVisitor(place))
    }
}
```

Reply:
Oh right, you don't have a visitor here so you can't directly do `deserialize_str`.

The advantage would've been that deserializing to an owned String requires copying and allocation. But it's fine.